src/java/picard/sam/PositionBasedDownsampleSam.java

```diff
@@ -0,0 +1,364 @@
+                out.addAlignment(rec);
+                kept++;
+            }
+            progress.record(rec);
+        }
+
+        out.close();
+
+        CloserUtil.close(in);
+    }
+
+    private void checkProgramRecords() {
+        final SamReader in = SamReaderFactory
+                .makeDefault()
+                .referenceSequence(REFERENCE_SEQUENCE)
+                .open(INPUT);
+
+        for (final SAMProgramRecord pg : in.getFileHeader().getProgramRecords()) {
+            if (pg.getProgramName() != null && pg.getProgramName().equals(PG_PROGRAM_NAME)) {
+
+                final String outText = "Found previous Program Record that indicates that this BAM has been downsampled already with this program. Operation not supported! Previous PG: " + pg.toString();
+
+                if (ALLOW_MULTIPLE_DOWNSAMPLING_DESPITE_WARNINGS) {
+                    log.warn(outText);
+                } else {
+                    log.error(outText);
+                    throw new PicardException(outText);
+                }
+            }
+        }
+        CloserUtil.close(in);
+    }
+
+    // scan all the tiles and find the smallest and largest coordinate (x & y) in that tile.
+    private void fillTileMinMaxCoord() {
+
+        final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
+
+        final ProgressLogger progress = new ProgressLogger(log, (int) 1e7, "Read");
+
+        int total = 0;
+
+        for (final SAMRecord rec : in) {
+            if (STOP_AFTER != null && total >= STOP_AFTER) break;
+
+            total++;
+            progress.record(rec);
+            final PhysicalLocation location = getSamRecordLocation(rec);
+
+            //Defaulting map will create a new Coord if it's not there.
+
+            final Coord Pos = tileCoord.get(location.getTile());
+
+            Pos.maxX = Math.max(Pos.maxX, location.getX());
+            Pos.minX = Math.min(Pos.minX, location.getX());
+
+            Pos.maxY = Math.max(Pos.maxY, location.getY());
+            Pos.minY = Math.min(Pos.minY, location.getY());
+
+            Pos.count++;
+
+        }
+
+        // now that we know what the maximal/minimal numbers were, we should increase/decrease them a little, to account for sampling error
+        for (final Coord coord : tileCoord.values()) {
+
+            final int diffX = coord.maxX - coord.minX;
+            final int diffY = coord.maxY - coord.minY;
+
+            coord.maxX += diffX / coord.count;
+            coord.minX -= diffX / coord.count;
+
+            coord.maxY += diffY / coord.count;
+            coord.minY -= diffY / coord.count;
+        }
+
+        CloserUtil.close(in);
+    }
+
+    private PhysicalLocation getSamRecordLocation(final SAMRecord rec) {
+        final PhysicalLocation pos = new PhysicalLocation();
+        opticalDuplicateFinder.addLocationInformation(rec.getReadName(), pos);
+        return pos;
+    }
+
+    private class CircleSelector {
```

Comment:
The problem is that you need to take a fraction of the area, AND a fraction of both the top and bottom edges of the tile. AND you have to be symmetric top/bottom and left/right (on the edges). This can still be done with an offset rectangle, but there are _more_ degrees of freedom, and thus more decisions to make. Also, the circle has the highest ratio of area to circumference, which is also being optimized here...happy to discuss this further if you like.